src/main/java/org/json/JSONObject.java

```diff
@@ -2417,17 +2424,19 @@ protected static Number stringToNumber(final String val) throws NumberFormatExce
                     }
                 }
             }
+            val = removeLeadingZerosOfNumber(input);
+            initial = val.charAt(0);
             // block items like 00 01 etc. Java number parsers treat these as Octal.
```

Comment:
let's remove this incorrect comment about Octal as well.

Reply:
Right, now that we are taking care of zeros, there isn't a chance of interpreting as Octal.